packages/caliper-core/lib/common/utils/caliper-utils.js

```diff
@@ -539,6 +539,21 @@ class CaliperUtils {
 
     }
 
+    /**
+     * Check if a named package is installed and accessible by caliper
+     * @param {string} packageName the name of the package to check
+     * @returns {boolean} boolean value for existence of accessible package
+     */
+    static packageIsInstalled(packageName) {
+        try {
+            require.resolve(packageName);
```

Comment:
This will tell whether the core package can see the queried package, not whether the caller can see it. This could lead to false reports for the caller. I overcame this by passing a `requireFunc` as an additional parameter, which defaults to `require` (of the caliper-utils module). This shouldn't be a problem for the internal packages (since they include the core deps), but 3rd party packages can run into problems when using this method.